src/Microsoft.Framework.Configuration.Json/JsonConfigurationSource.cs

```diff
@@ -78,7 +79,23 @@ public override void Load()
         internal void Load(Stream stream)
         {
             JsonConfigurationFileParser parser = new JsonConfigurationFileParser();
-            Data = parser.Parse(stream);
+            try
+            {
+                Data = parser.Parse(stream);
+            }
+            catch(JsonReaderException e)
+            {
+                // We are splitting error message based on e.Path and then getting substring of the error message
+                // before the last period(.)
+                // Sample e.Message : Unexpected end of content while loading JObject. Path 'address', line 7, 
+                // position 44.
+                // e.Path for above Message : address
+                var splitErrorMessageArray = e.Message.Split(new string[] { e.Path }, StringSplitOptions.None);
+
+                var errorMessage = splitErrorMessageArray[0].Substring(0, splitErrorMessageArray[0].LastIndexOf('.'));
```

Comment:
Parsing the message in any way is going to be too unreliable with localization. We'll have to settle for information already available on JsonReaderException properties.